frontend/simple/views/components/Transitions/MaskToModal/MaskToModal.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <component :is="tag">
+    <slot></slot>
+  </component>
+</template>
+<script>
+export default {
+  name: 'MaskToModal',
+  props: {
+    tag: {
+      type: String,
+      default: 'div'
+    }
+  },
+  data () {
+    return {
+      config: {
+        MaskToModal: {
+          // Store the dimensions and position of an element (Trigger or Target)
+          // so that Masker can access them when it needs to transform
+          elementsSpecs: {}, // a Trigger or Target
+          updateSpecsOf (el, elementId) {
+            const { width, height, top, left } = el.getBoundingClientRect()
+
+            this.elementsSpecs[elementId] = { width, height, top, left }
+          }
+        }
+      }
+    }
+  },
+  provide () {
+    return {
+      MaskToModal: this.config.MaskToModal
+    }
```

Comment:
This is fascinating. You figured out how to share the ephemeral data of a parent component with arbitrary child components in an instance-based manner using raw Vue components. 😮 👍

Reply:
Eheheh thanks! This is very similar to React's context API, so I just needed to dig into Vue docs until finding a similar future.
I already used it before too in the `Menu` component :)